aiohttp_devtools/runserver/watch.py

```diff
@@ -119,10 +121,35 @@ def _start_dev_server(self) -> None:
         self._process = Process(target=serve_main_app, args=(self._config, tty_path))
         self._process.start()
 
-    def _stop_dev_server(self) -> None:
+    async def _stop_dev_server(self) -> None:
         if self._process.is_alive():
             logger.debug('stopping server process...')
+            if self._config.shutdown_by_url:  # Workaround for signals not working on Windows
+                url = 'http://localhost:{}{}/shutdown'.format(self._config.main_port, self._config.path_prefix)
+                logger.debug('attempting to stop process via shutdown endpoint {}'.format(url))
+                try:
+                    # these errors are expected because the request kills the server
+                    with suppress(ServerDisconnectedError, ClientConnectorError, ClientOSError):
+                        async with ClientSession() as session:
+                            async with session.get(url):
+                                pass
+                except (ConnectionError, ClientError, asyncio.TimeoutError) as ex:
+                    if self._process.is_alive():
+                        logger.warning("shutdown endpoint caused an error (will try signals next): {} {}"
```

Comment:
I'm a little perplexed why coverage says we're getting to here. I'm not sure what tests are causing it.

Reply:
That is indeed strange - unfortunately since pytest swallows all the output from the tests, I can't grep them for "shutdown endpoint caused an error", so I'm not sure how to best track this down (without making all of the tests verbose at least).